Solutions/Branwen/GUI.cs

```diff
@@ -9,324 +9,460 @@
+		/// </summary>
+		/// <param name="sender"></param>
+		/// <param name="e"></param>
+		private void UseDBCheckBox_CheckedChanged(object sender, EventArgs e)
+		{
+			SetEnabledControls();
+		}
+
+		#endregion
+
+		#region Database Methods
+
+		/// <summary>
+		/// Writes all files in the Directory to the Databbase
+		/// </summary>
+		/// <param name="files"></param>
+		private void WriteDirectoryToDatabase(IEnumerable<BranwenFileInfo> files, MySqlConnection mySqlConnection)
+		{
+			string insertFilesStatement = "INSERT INTO SaurutobiMediaFiles (FileName, Extension, Size, MediaType, MediaDrive) VALUES ";
+			string insertPathsStatement = "INSERT INTO SaurutobiMediaPaths (FileName, PartOfPath, Path, MediaDrive) VALUES ";
+
+			bool firstFileVal = true;
+			foreach (BranwenFileInfo file in files)
+			{
+				if (!firstFileVal)
+				{
+					insertFilesStatement += ", ";
+					insertPathsStatement += ", ";
+				}
+				else
+				{
+					firstFileVal = false;
+				}
+
+				//File Name
+				insertFilesStatement += "('" + file.Name.Replace("'", "''") + "', ";
+
+				//File Extension
+				insertFilesStatement += "'" + file.Extension + "', ";
+
+				//File Size
+				insertFilesStatement += file.FileSize + ", ";
+
+				//Media Type
+				insertFilesStatement += "'" + file.Path[1].Replace("'", "''") + "', ";
+
+				//MediaDrive
+				insertFilesStatement += file.MediaDrive + ")";
+
+				bool firstPathVal = true;
+				for (int i = 0; i < file.Path.Count; i++)
+				{
+					if (!firstPathVal)
+					{
+						insertPathsStatement += ", ";
+					}
+					else
+					{
+						firstPathVal = false;
+					}
+					//File Name
+					insertPathsStatement += "('" + file.Name.Replace("'", "''") + "', ";
+
+					//PartOfPath
+					insertPathsStatement += i + ", ";
+
+					//Path
+					insertPathsStatement += "'" + file.Path[i].Replace("'", "''") + "', ";
+
+					//MediaDrive
+					insertPathsStatement += file.MediaDrive + ")";
+				}
+
+				fileCount++;
+			}
+			insertFilesStatement += ";";
+			insertPathsStatement += ";";
+
+			MySqlCommand insertCommand = new MySqlCommand(insertFilesStatement, mySqlConnection);
+			insertCommand.ExecuteNonQuery();
+			insertCommand = new MySqlCommand(insertPathsStatement, mySqlConnection);
+			insertCommand.ExecuteNonQuery();
+		}
+
+		/// <summary>
+		/// Writes an Excel document with all the contentes of the database
+		/// </summary>
+		/// <param name="mySqlConnection"></param>
+		/// <param name="outputFile"></param>
+		private void ExportDatabaseToFile(MySqlConnection mySqlConnection, string outputFile)
+		{
+			//Get all the MediaTypes
+			List<string> mediaTypes = new List<string>();
+			MySqlCommand command = new MySqlCommand("SELECT MediaType FROM SaurutobiMediaFiles GROUP BY MediaType;", mySqlConnection);
+			using (MySqlDataReader mediaTypeReader = command.ExecuteReader())
+			{
+				while (mediaTypeReader.Read())
+				{
+					mediaTypes.Add(mediaTypeReader["MediaType"].ToString());
+				}
+				mediaTypeReader.Close();

```

Comment:
again I wanna be super sure:P

On Thu, Dec 17, 2015 at 11:22 PM, Alex Ayerdi <notifications@github.com>
wrote:

> In Solutions/Branwen/GUI.cs
> <https://github.com/LbKStudios/Branwen/pull/5#discussion_r48000250>:
>
> > +		/// Writes an Excel document with all the contentes of the database
> > +		/// </summary>
> > +		/// <param name="mySqlConnection"></param>
> > +		/// <param name="outputFile"></param>
> > +		private void ExportDatabaseToFile(MySqlConnection mySqlConnection, string outputFile)
> > +		{
> > +			//Get all the MediaTypes
> > +			List<string> mediaTypes = new List<string>();
> > +			MySqlCommand command = new MySqlCommand("SELECT MediaType FROM SaurutobiMediaFiles GROUP BY MediaType;", mySqlConnection);
> > +			using (MySqlDataReader mediaTypeReader = command.ExecuteReader())
> > +			{
> > +				while (mediaTypeReader.Read())
> > +				{
> > +					mediaTypes.Add(mediaTypeReader["MediaType"].ToString());
> > +				}
> > +				mediaTypeReader.Close();
>
> connection is closed upon block end, no need for this line
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LbKStudios/Branwen/pull/5/files#r48000250>.
>